utils/internal/pre-publish-utils.ts

```diff
@@ -33,58 +37,56 @@ const checkPublishConfig = ({
     ) {
-        console.error(
-            `ERROR: ${name} must have a "prepublishOnly" script that runs "utils/package-pre-publish-check.sh".`,
+        error(
+            `${name} must have a "prepublishOnly" script that runs "utils/package-pre-publish-check.sh".`,
         );
         returnCode = false;
     }
+
     return returnCode;
 };
 
-const checkField = (pkgJson, field, value): boolean => {
-    let returnCode = true;
-    if (Array.isArray(value)) {
-        if (!value.includes(pkgJson[field])) {
-            console.error(
-                `ERROR: ${
-                    pkgJson.name
-                } must have a "${field}" set to one of ${value
-                    .map((value) => JSON.stringify(value))
-                    .join(", ")}.`,
-            );
-            returnCode = false;
-        }
-    } else if (pkgJson[field] !== value) {
-        console.error(
-            `ERROR: ${
-                pkgJson.name
-            } must have a "${field}" set to ${JSON.stringify(value)}.`,
+const checkExport = ({name, exports}, exportField, value): boolean => {
+    if (exports["."][exportField] !== value) {
+        error(
+            `${name} must include exports["."]["${exportField}"] set to ${JSON.stringify(value)}.\n    (eg. "exports": { ".": { "${exportField}": "${value}" } })`,
         );
-        returnCode = false;
+        return false;
     }
-    return returnCode;
-};
-
-const checkMain = (pkgJson): boolean =>
-    checkField(pkgJson, "main", "dist/index.js");
 
-const checkModule = (pkgJson): boolean =>
-    checkField(pkgJson, "module", "dist/es/index.js");
-
-const checkSource = (pkgJson): boolean =>
-    checkField(pkgJson, "source", ["src/index.js", "src/index.ts"]);
+    return true;
+};
 
-const checkPrivate = (pkgJson): boolean => {
+export const checkPrivate = (pkgJson): boolean => {
     if (pkgJson.private) {
         console.warn(
             `${pkgJson.name} is private and won't be published to NPM.`,
         );
-        return true;
+        return false;
     }
-    return false;
+    return true;
 };
 
-const checkEntrypoints = (pkgJson): boolean =>
-    checkModule(pkgJson) && checkMain(pkgJson);
+export const checkExports = (pkgJson): boolean => {
+    if (pkgJson.exports == null) {
+        error(`${pkgJson.name} must have a "exports" field.`);
+        return false;
+    }
+
+    if (pkgJson.exports["."] == null) {
+        error(
+            `${pkgJson.name} must have an "exports" field named "." (eg. "exports": { ".": {...} }).`,
+        );
+        return false;
+    }
 
-export {checkPublishConfig, checkEntrypoints, checkSource, checkPrivate};
+    let returnValue = true;
+    returnValue =
+        checkExport(pkgJson, "import", "./dist/index.js") && returnValue;
+    returnValue =
+        checkExport(pkgJson, "types", "./dist/index.d.ts") && returnValue;
+    returnValue =
+        checkExport(pkgJson, "source", "./src/index.ts") && returnValue;
+
+    return returnValue;
```

Comment:
I feel like it would be clearer and more concise to just use multiple variables.

```suggestion
    const hasTranspiled = checkExport(pkgJson, "import", "./dist/index.js");
    const hasTypes = checkExport(pkgJson, "types", "./dist/index.d.ts");
    const hasSrc = checkExport(pkgJson, "source", "./src/index.ts");

    return hasTranspiled && hasTypes && hasSrc;
```